### PR TITLE
[CSL 3407] Add support for pre-filter expression for Recommendations

### PR DESCRIPTION
**Pull request by jjl014**
### Updates:
* Add pre-filter support to Recommendations
* Add test

### CODE REVIEW

**Review by Herfaifman — APPROVED** [review]

**Review by HFaifman — APPROVED** [review]
Code-wise it looks great. Let me know if it needs to be manually tested as well and I´ll whip it up locally.

File: src/modules/recommendations.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable object-curly-newline, no-param-reassign */
+/* eslint-disable object-curly-newline, no-param-reassign, no-underscore-dangle */
```
Comment (esezen) [inline src/modules/recommendations.js:1]: Do we still need this `no-underscore-dangle`? I'm guessing you removed some code that needed this

**Review by esezen — APPROVED** [review]
LGTM! Left one comment about the lint rule override

File: src/modules/recommendations.js (hunk 1 above)
Comment (jjl014) [reply src/modules/recommendations.js]: Thanks for catching this! I think I added it for something else and then removed it. Will be removing the rule override 👍

**Pull request merged**